### ISSUE TITLE
cem-plugin-expanded-types: built-in TS utility types not expanded

### ISSUE THREAD

**Issue by certainlyakey** [break-stuff/cem-tools#107]
Hi! 
I noticed that utility types of Typescript (such as `Exclude`) don't seem to work with cem-plugin-expanded-types. If you use them with a union type (regardless whether imported or local to a file), nothing gets expanded in `custom-elements.json`.

As a quick example, a type `MySize` works and gets expanded (with `'sm' | 'md' | 'lg'` as a value), but type Exclude<MySize, 'lg'> does not.

My version is the latest one at the moment (1.2.0).

Here's a [small repro](https://github.com/certainlyakey/cem-plugin-expanded-types-union-type-repro) with a couple of commits demonstrating [different](https://github.com/certainlyakey/cem-plugin-expanded-types-union-type-repro/commit/b0811255e550e9cb021594b7261955afaff9cc3e) [utility types](https://github.com/certainlyakey/cem-plugin-expanded-types-union-type-repro/commit/e8389a5af03affcab0a39e01e170d9d58aea0238) that don't work.

**Comment by break-stuff** [comment]
Yeah, unfortunately, these and other generics do not work very well. I haven't been able to figure out how to resolve those yet.

**Comment by break-stuff** [comment]
@certainlyakey  I think this has been resolved in my new package that you can find here:
https://wc-toolkit.com/documentation/type-parser/

**Comment by certainlyakey** [comment]
Thanks, I will definitely try it out!Sent from my iPhoneOn 17. Feb 2025, at 23:21, Burton Smith ***@***.***> wrote:﻿
  @certainlyakey  I think this has been resolved in my new package that you can find here:
https://wc-toolkit.com/documentation/type-parser/—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>



break-stuff left a comment (break-stuff/cem-tools#107)
@certainlyakey  I think this has been resolved in my new package that you can find here:
https://wc-toolkit.com/documentation/type-parser/

**Comment by break-stuff** [comment]
@certainlyakey  Closing as this has moved to the [WC Toolkit](https://wc-toolkit.com/documentation/type-parser/). Please let me know if you're still experiencing issues.
